build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -7,9 +7,7 @@
 # Utility function to return the platform name in crosscompile.
 function(iree_get_platform PLATFORM)
   if(ANDROID AND CMAKE_ANDROID_ARCH_ABI STREQUAL "arm64-v8a")
-    set(_PLATFORM "android-arm64-v8a")
-  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-    set(_PLATFORM "x86_64")
+    set(_PLATFORM "arm64-v8a-Android")
```

Comment:
How is this related?

Reply:
The original ad-hoc naming is to fit the names in old benchmark suites. I canonicalized them since we don't need to match old benchmark suites anymore.

Reply:
I guess I'm just confused why android is still special but also gets a new name. Does it not set `CMAKE_SYSTEM_PROCESSOR` and `CMAKE_SYSTEM_NAME`? Maybe we should do that

Reply:
android.toolchain.cmake use its own variables, although they are supported by CMake.
https://android.googlesource.com/platform/ndk/+/master/build/cmake/android.toolchain.cmake#116

Reply:
It looks like CMake has an internal module to guess the `CMAKE_SYSTEM_PROCESSOR` for Android
https://github.com/Kitware/CMake/blob/master/Modules/Platform/Android-Determine.cmake
But for arm64-v8a, CMake will set `CMAKE_SYSTEM_PROCESSOR` as `aarch64`.

Reply:
Simplified to just use `${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}`. We can revise this if there is an issue to use `aarch64` in CMake.